API/models/model.go

```diff
@@ -124,6 +124,29 @@ func PropagateParentIdChange(ctx context.Context, oldParentId, newId string, ent
 	return nil
 }
 
+// PropagateDomainChange: search for all objects with reference to the modified domain
+func PropagateDomainChange(ctx context.Context, oldDomainId, newDomainId string) error {
+	// Find all objects containing this domain
+	req := bson.M{"domain": primitive.Regex{Pattern: "^" + oldDomainId + "(\\" + u.HN_DELIMETER + "|$)", Options: ""}}
+	// For each object found, replace old domain by new
+	update := bson.D{{
+		Key: "$set", Value: bson.M{
+			"domain": bson.M{
+				"$replaceOne": bson.M{
+					"input":       "$domain",
+					"find":        oldDomainId,
+					"replacement": newDomainId}}}}}
+	for i := u.STRAYOBJ + 1; i <= u.GROUP; i++ {
+		_, e := repository.GetDB().Collection(u.EntityToString(i)).UpdateMany(ctx,
+			req, mongo.Pipeline{update})
```

Comment:
Why is a pipeline necessary if it is a single operation to be performed?

Reply:
There was a discussion about this with the other propagate function, if I still remember it right, when we use set in an aggregation pipeline it works as an aggregator operator, if not, it's an update operator and they don't have the same behaviour

Reply:
https://www.mongodb.com/docs/manual/tutorial/update-documents-with-aggregation-pipeline/